plugins/module_utils/_global_catalog.py

```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
-                    "DFHRMUTL failed with RC 16 - {0}".format(stdout_with_rc[0]), executions
-                )
-            elif reason_code is None:
+    try:
+        for x in range(MVS_CMD_RETRY_ATTEMPTS):
+            dfhrmutl_response, jcl_executions = _execute_dfhrmutl(qualified_file_path)
+            dfhrmutl_rc = dfhrmutl_response.get("ret_code").get("code")
+
+            allContent = []
+            for ddname in dfhrmutl_response.get("ddnames"):
+                    allContent += ddname.get("content")
+            stdout_raw = "".join(allContent)
+
+            executions.append(jcl_executions)
+            executions.append(
+                _execution(
+                    name="DFHRMUTL - {0} - Run {1}".format(
+                        "Get current catalog" if cmd == "" else "Updating autostart override",
+                        x + 1),
+                    rc=dfhrmutl_rc,
+                    stdout=stdout_raw,
+                    stderr=dfhrmutl_response.get("ret_code").get("msg_txt", "")))
+            
+            if dfhrmutl_rc not in (0, 16):
                 raise MVSExecutionException(
-                    "DFHRMUTL failed with RC 16 but no reason code was found",
-                    executions,
-                )
+                    "DFHRMUTL failed with RC {0}".format(
+                        dfhrmutl_rc), executions)
+
+            if dfhrmutl_rc == 0:
+                break
+            if dfhrmutl_rc == 16:
+                formatted_stdout_lines = [
+                    "{0}".format(element.replace(" ", "").upper())
+                    for element in stdout_raw.split("\n")
+                ]
+                stdout_with_rc = list(filter(lambda x: "REASON:X" in x, formatted_stdout_lines))
+
+                reason_code = _get_reason_code(stdout_with_rc)
+                if reason_code and reason_code != "A8":
+                    raise MVSExecutionException(
+                        "DFHRMUTL failed with RC 16 - {0}".format(stdout_with_rc[0]), executions
+                    )
+                elif reason_code is None:
+                    raise MVSExecutionException(
+                        "DFHRMUTL failed with RC 16 but no reason code was found",
+                        executions,
+                    )
+
+        if cmd != "":
+            return executions
+    finally:
+        os.remove(qualified_file_path)
+    return executions, _get_catalog_records(stdout_raw)
 
-        else:
-            raise MVSExecutionException(
-                "DFHRMUTL failed with RC {0}".format(
-                    dfhrmutl_response.rc), executions)
 
-    if cmd != "":
-        return executions
+def _execute_dfhrmutl(rmutl_jcl_path, job_name="DFHRMUTL"):
+    executions = _submit_jcl(rmutl_jcl_path, job_name)
+    job_id = executions[0].get("stdout").strip()
 
-    return executions, _get_catalog_records(dfhrmutl_response.stdout)
+    #Give RMUTL a second to run
+    sleep(1)
```

Comment:
Can we make this an obvious constant instead of just the number here. I know we only use it once but would be good to make it obvious